core/server/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -134,6 +134,28 @@ public static void main(String[] args) {
   private long mStartTimeMs = -1;
 
   /**
+   * @return a list of the enabled master services' names
+   */
+  public static List<String> getNames() {
+    List<String> names = Lists.newArrayList();
```

Comment:
Also the code here has very similar pattern to the constructor where we create those specific masters. As a result, for each master, service loader is invoked twice: one for `create` a master and one for get the name. This is error-prune because any change to the masters need to be taken care at both places. Besides, invoking serviceloader in a getter method looks less ideal to me.

How about consolidate the logic only in constructor, and push the names into a member variable e.g. `mServiceNames`. In this getter method,  we can simply return that member variable. In that way, we only need to maintain one code block in both creating masters and recording their names.